### PR TITLE
native boards: Minor docs improvements

### LINKED ISSUE #65565: doc: native_sim SDL instructions in 'boards/posix/native_sim/doc/index'
**Describe the bug**

"On a Ubuntu 22.04 host system, for example, install the pkg-config and libsdl2-dev:i386 packages, and configure the pkg-config search path with these commands [...]"

I believe these instructions may need to be amended as I am guessing this might not work out of the box for most people. I had to do the following first:

```
sudo dpkg --add-architecture i386
sudo apt update
```

**Environment**

* Page: `boards/posix/native_sim/doc/index`
* Version: main
* SHA-1: 65fd7d42467225c3c9df013fb710587a62bbf874

### PR DESCRIPTION
**Pull request by aescolar**
boards native: Add not supported features
    
    Add a section with the most significant not supported features.

----

    native_sim: Improve instructions for i386 packages
    
    To the instructions that install "foreign architecture" (i386)
    packages on a x86-64 system, add the commands to ensure
    the architecture is added if it wasn't already.

Fixes: #65565

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

File: boards/posix/doc/arch_soc.rst [hunk 1]
```diff
@@ -169,6 +169,22 @@ In the previous example, modifying the code as follows would work:
     #endif
    }
 
+.. _posix_arch_not_supported:
+
+Significant not supported features
+**********************************
```
Comment (kartben) [inline boards/posix/doc/arch_soc.rst:175]:
```suggestion
Significant unsupported features
********************************
```

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks for this!

File: boards/posix/doc/arch_soc.rst [hunk 2]
```diff
@@ -169,6 +169,22 @@ In the previous example, modifying the code as follows would work:
...
+
+* :ref:`User mode/userspace <usermode_api>` . When you build for these targets,
+  :kconfig:option:`CONFIG_USERSPACE` will always be disabled,
+  and all calls into the kernel will be done as normal calls.
+
+* Stack checks: :kconfig:option:`CONFIG_HW_STACK_PROTECTION`,
+  :kconfig:option:`CONFIG_STACK_CANARIES`, and
+  :kconfig:option:`CONFIG_THREAD_ANALYZER`.
+  In the POSIX architecture, the threads' stacks allocated by Zephyr are not actually used as
+  the threads stacks.
```
[the remark this answers is not in the document]
Comment (aescolar) [reply boards/posix/doc/arch_soc.rst]: Hopefully it is a bit better now :)

**Comment by aescolar** [comment]
Thanks @kartben, updated with your feedback

File: boards/posix/doc/arch_soc.rst [hunk 3]
```diff
@@ -169,6 +169,23 @@ In the previous example, modifying the code as follows would work:
...
+Currently, these are the most significant features which are not supported in this architecture:
+
+* :ref:`User mode/userspace <usermode_api>`: When building for these targets,
+  :kconfig:option:`CONFIG_USERSPACE` will always be disabled,
+  and all calls into the kernel will be done as normal calls.
+
+* Stack checks: :kconfig:option:`CONFIG_HW_STACK_PROTECTION`,
+  :kconfig:option:`CONFIG_STACK_CANARIES`, and
+  :kconfig:option:`CONFIG_THREAD_ANALYZER`.
+  Due to Zephyr allocated threads' stacks not being used as in other architectures. Check
```
Comment (kartben) [inline boards/posix/doc/arch_soc.rst:186]: Better yes! It just feels odd that the sentence has no verb. LMK if you feel the below is better.

```suggestion
  This is due to how Zephyr allocated threads' stacks are not `actually` being used like they are in other architectures. Check
```

Comment (aescolar) [reply boards/posix/doc/arch_soc.rst]: Changed :)

**Review by kartben — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**